shipment_advice/tests/test_shipment_advice.py

```diff
@@ -246,3 +246,31 @@ def test_shipment_advice_draft(self):
     def test_shipment_name(self):
         self.assertTrue("OUT" in self.shipment_advice_out.name)
         self.assertTrue("IN" in self.shipment_advice_in.name)
+
+    def test_validate_shipment_advice_when_fully_done(self):
+        picking = self._prepare_picking_with_two_packages()
+        line1, line2 = picking.move_line_ids
+        # Load first package in shipment advice 2
+        pl1 = line1.package_level_id
+        self._in_progress_shipment_advice(self.shipment_advice_out)
+        self.assertEqual(self.shipment_advice_out.state, "in_progress")
+        self._load_records_in_shipment(self.shipment_advice_out, pl1)
+        pl2 = line2.package_level_id
+        # Load second package into shipment advice
+        self._load_records_in_shipment(self.shipment_advice_out, pl2)
+        # Assign planned move_ids that are assigned to test function
+        self.shipment_advice_out.planned_move_ids = line2.move_id + line1.move_id
+        self.assertEqual(line1.move_id.state, "assigned")
+        self.assertEqual(line2.move_id.state, "assigned")
+        self.assertEqual(self.shipment_advice_out.state, "in_progress")
+        # Validating if moves in "incorrect" states do not trigger _action_done()
+        self.shipment_advice_out.validate_when_fully_done()
+        self.assertEqual(self.shipment_advice_out.state, "in_progress")
+        # Validating if moves in one "incorrect" state do not trigger _action_done()
+        line1.move_id.state = "done"
+        self.shipment_advice_out.validate_when_fully_done()
+        self.assertEqual(self.shipment_advice_out.state, "in_progress")
+        line2.move_id.state = "done"
+        self.shipment_advice_out.validate_when_fully_done()
+        self.assertEqual(self.shipment_advice_out.state, "done")
+        self.assertEqual(picking.state, "done")
```

Comment:
In this test you should call the validate_when_full_done anymore.
I would be good to just validate the pickings. And afterwards check if the shipment advice is closed.